### ISSUE TITLE
`mas` doesn't build on ElCap with Xcode 8

### ISSUE THREAD

**Issue by chdiza** [Homebrew/homebrew-core#5182]
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?

`mas` failed to build on ElCap with Xcode 8.  Seems to be a problem with its gem installation phase, which is crapping out trying to find something in the 10.12 SDK:

```
~$ brew install mas
git config --local --replace-all homebrew.private
==> Using the sandbox
/usr/bin/sandbox-exec -f /var/folders/_n/whvqgz_s01bf90cz7k0615wr0000gn/T/homebrew20160924-81358-10k2pnh.sb nice /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/hb/Library/Homebrew -- /usr/local/hb/Library/Homebrew/build.rb /usr/local/hb/Library/Taps/homebrew/homebrew-core/Formula/mas.rb --verbose
==> Downloading https://github.com/argon/mas/archive/v1.3.0.tar.gz
Already downloaded: /stuff/brache/mas-1.3.0.tar.gz
==> Verifying mas-1.3.0.tar.gz checksum
tar xvzf /stuff/brache/mas-1.3.0.tar.gz
[Unpacking....]
==> gem install bundler
Successfully installed bundler-1.13.1
Parsing documentation for bundler-1.13.1
Installing ri documentation for bundler-1.13.1
1 gem installed
==> script/bootstrap
Fetching gem metadata from https://rubygems.org/.............
Fetching version metadata from https://rubygems.org/.
Rubygems 2.0.14.1 is not threadsafe, so your gems will be installed one at a time. Upgrade to Rubygems 2.1.0 or higher to enable parallel gem installation.
Installing i18n 0.7.0
Installing json 1.8.3 with native extensions
Installing minitest 5.8.4
Installing thread_safe 0.3.5
Installing claide 0.9.1
Installing colorize 0.7.7
Installing colored 1.2
Installing rouge 1.10.1
Using bundler 1.13.1
Gem::Installer::ExtensionBuildError: ERROR: Failed to build gem native
extension.

/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
extconf.rb
creating Makefile

make "DESTDIR="
compiling generator.c
In file included from generator.c:1:
In file included from ./../fbuffer/fbuffer.h:5:
In file included from
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby.h:33:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/include/ruby-2.0.0/ruby/ruby.h:24:10:
fatal error: 'ruby/config.h' file not found
#include "ruby/config.h"
         ^
1 error generated.
make: *** [generator.o] Error 1


Gem files will remain installed in
/private/var/folders/_n/whvqgz_s01bf90cz7k0615wr0000gn/T/mas-20160924-81360-1lx8e3b/mas-1.3.0/.gem/gems/json-1.8.3
for inspection.
Results logged to
/private/var/folders/_n/whvqgz_s01bf90cz7k0615wr0000gn/T/mas-20160924-81360-1lx8e3b/mas-1.3.0/.gem/gems/json-1.8.3/ext/json/ext/generator/gem_make.out

An error occurred while installing json (1.8.3), and Bundler cannot continue.
Make sure that `gem install json -v '1.8.3'` succeeds before bundling.
git config --local --replace-all homebrew.private
==> Formula
Path: /usr/local/hb/Library/Taps/homebrew/homebrew-core/Formula/mas.rb
==> Configuration
HOMEBREW_VERSION: 1.0.3-27-g210de29
ORIGIN: https://github.com/Homebrew/brew
HEAD: 210de2916b090cf7d5828cadbaab58e42a36a073
Last commit: 20 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 59ce30641366d0a6803f07f07058202722a95c77
Core tap last commit: 25 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/hb
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit ivybridge
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.10.0 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.11.6-x86_64
Xcode: 8.0
CLT: 7.3.1.0.1.1461711523
X11: 2.7.9 => /opt/X11
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
SDKROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
MACOSX_DEPLOYMENT_TARGET: 10.11
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/hb/Library/Homebrew/os/mac/pkgconfig/10.11
HOMEBREW_VERBOSE: 1
HOMEBREW_BUILD_FROM_SOURCE: 1
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /private/var/folders/_n/whvqgz_s01bf90cz7k0615wr0000gn/T/mas-20160924-81360-1lx8e3b/mas-1.3.0/.gem/bin:/usr/local/hb/Library/Homebrew/shims/super:/usr/bin:/bin:/usr/sbin:/sbin

Error: mas 1.3.0 did not build
Logs:
     /Users/chdiza/Library/Homebrew/mas/01.gem
     /Users/chdiza/Library/Homebrew/mas/02.bootstrap
```

**Comment by ilovezfs** [comment]
I can't reproduce the issue
```
** BUILD SUCCEEDED **

==> Cleaning
Fixing /usr/local/Cellar/mas/1.3.0/bin/mas permissions from 755 to 555
==> Finishing up
ln -s ../Cellar/mas/1.3.0/bin/mas mas
==> Summary
🍺  /usr/local/Cellar/mas/1.3.0: 4 files, 6.2M, built in 37 seconds
```

gist-logs?

**Comment by chdiza** [comment]
The logs contain nothing that wasn't already posted above.

**Comment by ilovezfs** [comment]
you may want to try in a new user account

**Comment by chdiza** [comment]
Made no difference, alas.  Same exact error.

**Comment by ilovezfs** [comment]
Have you turned SIP off by any chance?

**Comment by chdiza** [comment]
Yeah, I have.  It's odd that that should be the issue.  Is it known to cause accusations of missing libraries?

I'll try building `mas` again when I get a chance to reboot and turn the Apple Chastity Belt back on. :)

**Comment by ilovezfs** [comment]
@chdiza lol. I'm just trying to figure out how many degrees of freedom there are to the mystery. If a file was moved, then turning back on SIP wouldn't change the situation.

**Comment by ilovezfs** [comment]
@chdiza what's the output of
```
xcode-select -p
```

**Comment by chdiza** [comment]
> what's the output of xcode-select -p

`/Applications/Xcode.app/Contents/Developer`

(I'm kinda surprised that `brew --config` doesn't show this, at least not explicitly.)

**Comment by chdiza** [comment]
This no longer happens with the newly-available `mas` 1.3.1.

**Comment by ilovezfs** [comment]
bizarre.
